jd4/case.py

```diff
@@ -256,20 +256,53 @@ def read_legacy_cases(config, open):
                           memory_bytes,
                           int(score_str))
 
-def read_yaml_cases(config, open):
-    for case in yaml.safe_load(config)['cases']:
-        if 'judge' not in case:
-            yield DefaultCase(partial(open, case['input']),
-                              partial(open, case['output']),
-                              parse_time_ns(case['time']),
-                              parse_memory_bytes(case['memory']),
-                              int(case['score']))
-        else:
-            yield CustomJudgeCase(partial(open, case['input']),
+def read_auto_cases(open, zip_file, time_limit='1s', memory_limit='128m'):
+    cnt = 0
+    files = zip_file.namelist()
+    prefix = ''
+    cases = []
+    re_input = re.compile(r'([a-zA-Z]*)([0-9]*)\.in')
+    for i in files:
+        match = re_input.match(i)
+        if match:
+            prefix = match.group(1)
+            cases.append(match.group(2))
+            cnt += 1
+    if cnt == 0:
+        raise FormatError('No testdata found.')
+    cases.sort()
+    for i in cases:
+        yield DefaultCase(partial(open, prefix + str(i) + '.in'),
+                          partial(open, prefix + str(i) + '.out'),
+                          parse_time_ns(time_limit),
+                          parse_memory_bytes(memory_limit),
+                          int(100//cnt))
+
+def read_yaml_cases(config, open, zip_file):
+    try:
```

Comment:
You can just remove the try-except block. `yaml.safe_load` will throw an exception it self.